src/client.rs

```diff
@@ -0,0 +1,41 @@
+use crate::error::Error;
+use prost_types::Any;
+
+pub struct Client(internal::client::DaprClient<tonic::transport::Channel>);
+
+impl Client {
+    /// Connect to a Dapr enabled app.
+    pub fn connect(addr: String) -> Result<Self, Error> {
+        Ok(Client(internal::client::DaprClient::connect(addr)?))
+    }
+
+    /// Invoke a method in a Dapr enabled app.
+    pub async fn invoke_service<I, M>(
+        &mut self,
+        app_id: I,
+        method_name: M,
+        data: Option<Any>,
+    ) -> Result<InvokeServiceResponseEnvelope, Error>
```

Comment:
It should be `Result<(Option<Any>, Metadata), Error>`, if we want to hide the generated proto binding.

Or, at least use a typedef to hide the `xxxEvelope` suffix.

```rust
pub type InvokeServiceResponse = internal::InvokeServiceResponseEnvelope
```

Reply:
Yea, I personally don't mind leaking the generated type. I agree the type def could be nice since the name is better. We could also more easily provide documentation for it then.